proposals/docker_build.md

```diff
@@ -0,0 +1,45 @@
+# Proposal: Build to Docker Daemon
+
+## Motivation
+
+Currently, Jib builds and pushes container images to a Docker registry without the need for a Docker daemon. However, for development use cases where the developer does not have a registry set up but do have a Docker daemon available, they may wish to build to the Docker daemon directly.
+
+## Goals
+
+* Build to local Docker daemon
+* Build to [minikube](https://github.com/kubernetes/minikube) (remote) Docker daemon
+* Minimize extra configuration
+
+## Constraints
+
+* Building to local Docker daemon should be an extra feature, not the default
+* Should not depend on a Docker CLI
+
+## Intended Workflow
+
+Building to a Docker daemon should be as simple as calling the build task/goal (`jib:build` for Maven and `jib` for Gradle) with an additional parameter called `buildToDockerDaemon` enabled.
```

Comment:
I wonder if we should just add a new task?

jib:buildToDockerDaemon?

Reply:
and if we do this, perhaps we can match the bazel dockerless builds and also produce an intermediate tar, which may or may not help us with simple skaffold integration?

Reply:
Yeah, a new task does sound good so that we wouldn't need any extra configuration to clutter the default build+push workflow. And yeah, exporting the intermediate tar sounds good. Maybe we can instead do:

`jib:tar` to build the tar
`jib:load` to load the tar into the Docker daemon

This way, we could implement `jib:tar` first to enable integration with Skaffold.

Reply:
Decided on `jib:buildDocker`